apis/vshn/v1/common_types.go

```diff
@@ -9,6 +9,11 @@ type K8upBackupSpec struct {
 	Schedule string `json:"schedule,omitempty"`
 
 	Retention K8upRetentionPolicy `json:"retention,omitempty"`
+
+	// DeletionProtection will protect the instance's data from being deleted while the composite or claim still exists.
+	// This is enabled by default.
+	// +kubebuilder:default=true
+	DeletionProtection *bool `json:"deletionProtection,omitempty"`
```

Comment:
Yes. Having multiple deletion protection flags will only be confusing for the customer.

Although the only service that currently has that is PostgreSQL and that implementation is not very good. I'd like to have the webhook based deletion protection for the claims rather sooner than later. I don't want to wait with that until we have an incident...

Reply:
Why do we have to expose this Deletion Protection to the customer, this level of protection with webhooks should be abstracted away from customer all together, don't you thing so?

Reply:
An override to temporarily disable the backend deletion protection will be useful. For example if we need to reinitialize Galera Clusters.

Having an override on the claim inevitably exposes it to the customer. So my though was, that we put it under the same key that the customer can use to control the frontend deletion protection in the future.

This makes it easier to understand and leaves less room for miss configuration. The protection is either on or off.

Reply:
This approach means that we do not have our base data loss protection when the customer disables deletion protection. This is wrong in my opinion, the customer would not want in any case that his data is less protected even when he disables front end deletion protection. I would not expose the backend deletion protection to the customer and would enable it by default all the time.

Reply:
> I would not expose the backend deletion protection to the customer and would enable it by default all the time.

The problem here is: it could get pretty messy to have an override for the deletion protection that is not exposed to the customer as well. See the billing override. I would rather not have that much complexity for that, as it could increase the chances of failure.

Reply:
Let's discuss that during the daily. I feel I am missing something here.